index.html

```diff
@@ -5,45 +5,364 @@
+{"num":38,"neighborhood":"Shipley Terrace","poverty_rate":38,"avg_family_income":38427},
+{"num":33,"neighborhood":"Capitol View","poverty_rate":22,"avg_family_income":53239},
+{"num":23,"neighborhood":"Ivy City","poverty_rate":32,"avg_family_income":48388},
+{"num":39,"neighborhood":"Bellevue","poverty_rate":36,"avg_family_income":41605},
+{"num":9,"neighborhood":"Southwest","poverty_rate":19,"avg_family_income":94339},
+{"num":22,"neighborhood":"Brookland","poverty_rate":23,"avg_family_income":81931},
+{"num":1,"neighborhood":"Adams Morgan","poverty_rate":5.6,"avg_family_income":162891},
+{"num":37,"neighborhood":"Barry Farm","poverty_rate":41,"avg_family_income":38015},
+{"num":8,"neighborhood":"Downtown","poverty_rate":26,"avg_family_income":142881},
+{"num":35,"neighborhood":"Naylor Gardens","poverty_rate":17,"avg_family_income":72747},
+{"num":6,"neighborhood":"DuPont Circle","poverty_rate":13,"avg_family_income":177893},
+{"num":20,"neighborhood":"Michigan Park","poverty_rate":11,"avg_family_income":87062},
+{"num":21,"neighborhood":"Edgewood","poverty_rate":17,"avg_family_income":91520},
+{"num":2,"neighborhood":"Columbia Heights","poverty_rate":17,"avg_family_income":87204},
+{"num":7,"neighborhood":"Logan Circle","poverty_rate":12,"avg_family_income":123697},
+{"num":3,"neighborhood":"Shaw","poverty_rate":17,"avg_family_income":101583},
+{"num":27,"neighborhood":"Navy Yard","poverty_rate":34,"avg_family_income":77952}];
+
+var MAX_INCOME = 304995;
+var MAX_POVERTY = 45;
+
+var options = {
+  sorted: false,
+  filtered: false,
+  page:0
+};
+
+
+
+
+function getNumberOfPages() {
+  return Math.ceil(dc_info.length/numberPerPage);
+}
+
+function nextPage() {
+  currentPage += 1;
+  updateTable();
+}
+
+//Sort by neighborhood name and poverty rate in descending order
+
+function compareStrings (a,b) {
+  a = a.toLowerCase();
+  b = b.toLowerCase();
+  return (a < b) ? -1 : (a > b) ? 1: 0;
+}
+dc_info.sort(function (a, b) {
+  return compareStrings(a.neighborhood, b.neighborhood);
+});
+
+//Sort by income and poverty rate
+
+
+function updateTable() {
+
+  var data = dc_info.slice()
+
+
+//Filter by Poverty>20%
+  if (options.filtered) {
+     if (options.filtered === 'filter-p') {
+        data = data.filter(function (a) { 
+          return a.poverty_rate > 20; });
+      }
+}
+//Filter by 6 figure income
+  if (options.filtered) {
+    if (options.filtered === 'filter-i'){
+      data = data.filter(function (a) {
+        return a.avg_family_income > 99999; });
+    }
+}
+
+//Sort
+
+if (options.sorted) {
+    if (options.sorted === 'desc') {
+      data.sort(function (a, b) { return b.avg_family_income - a.avg_family_income; });
+    } else if (options.sorted === 'desc-poverty') {
+      data.sort(function (a, b) { return b.poverty_rate - a.poverty_rate; });
+    }
+ }
+
+//My attempt at Pagination for Next button, not working though
+
+
+
+  if (options.page) {
```

Comment:
One gotcha to look out for, that could end up being a source of bugs, is that 0 evaluates to false in Javascript. So if the default page is 0, this code won't run. Yeah, it's one of the more annoying aspects of Javascript.

However, if we can count on options.page always being a number, we don't actually need to check this. We just need to use whatever number is given us to show the correct set of results (and the math you have on line 271 is exactly right). Of course, we also need to handle things like "What if the requested page is 2, and we only have one page of results?" Or, if there's a previous page button that decrements options.page by 1, "What if the requested page is a negative number?"